processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryRunnerFactory.java

```diff
@@ -146,7 +147,9 @@ public SegmentMetadataQueryRunnerFactory(
                           @Override
                           public Sequence<SegmentAnalysis> call() throws Exception
                           {
-                            return input.run(query, responseContext);
```

Comment:
for other query types, ChainedExecutionQueryRunner takes care this by materializing the result, can we use that here also ?